core/cap-0038.md

```diff
@@ -689,21 +690,26 @@ if isAuthorizedToMaintainLiabilities(tl) && !isAuthorizedToMaintainLiabilities(e
 
         //redeem lp's pool shares using the logic from LiquidityPoolWithdrawOp
         assetsAndAmounts = redeem(poolTL)
-
-        foreach (assetAndAmount in assetsAndAmounts)
-            tlA = loadTrustLine(trustor, assetAndAmount.asset)
-            if tlA.availableLimit < amountA
-                ClaimableBalance cb
-                cb.amount = assetAndAmount.amount
-                cb.asset = assetAndAmount.asset
-                cb.claimant = trustor
-
-                if isClawbackEnabledOnTrustline(tl)
-                    cb.flags = CLAIMABLE_BALANCE_CLAWBACK_ENABLED_FLAG
-                
-                create(cb)
-            else
-                tlA.balance += assetAndAmount.amount
+        
+        //free up reserves from poolTL for the claimable balances below
+        delete(poolTL)
+
+        foreach (assetAndAmount in assetAndAmounts)
+            tlA = loadTrustLine(trustor, assetsAndAmount.asset)
+            
+            LedgerEntry le;
+            le.sponsoringID = trustor
```

Comment:
I don't think this is right in the case that the pool share trust line was itself sponsored?

Reply:
Good point. Fixed.